src/pm/hydra/tools/bootstrap/external/slurm_query_node_list.c

```diff
@@ -92,8 +92,8 @@ static HYD_status list_to_nodes(char *str)
     /* compile element regex for old format: "h14" */
     regcomp(&ematch_old, "([a-zA-Z]+[0-9]+)", REG_EXTENDED);
 
-    /* compile group-0 regex for new format: "h-[00-12,14] | h-14" */
-    regcomp(&gmatch_new[0], "(,|^)([a-zA-Z]+[0-9]*-)(\\[[-,0-9]+\\]|[0-9]+)(,|$)", REG_EXTENDED);
+    /* compile group-0 regex for new format: "h-[00-12,14] | h[00-12,14] | h-14" */
+    regcomp(&gmatch_new[0], "(,|^)([a-zA-Z]+[0-9]*-?)(\\[[-,0-9]+\\]|[0-9]+)(,|$)", REG_EXTENDED);
```

Comment:
```
The Internet standards (Requests for Comments) for protocols mandate that component hostname labels
may contain only the ASCII letters 'a' through 'z' (in a case-insensitive manner), the digits '0'
through '9', and the hyphen ('-'). The original specification of hostnames in RFC 952, mandated that
labels could not start with a digit or with a hyphen, and must not end with a hyphen. However, a
subsequent specification (RFC 1123) permitted hostname labels to start with digits. No other
symbols, punctuation characters, or white space are permitted.
```

I don't think your format above is allowing this.

Reply:
So `gmatch_new[0]` should be compiled with `(,|^)([a-zA-Z0-9]+-?)(\\[-,0-9]+\\] | [0-9]+)(,|$)` instead